src/crimson/osd/replicated_recovery_backend.cc

```diff
@@ -463,7 +460,7 @@ ObjectRecoveryInfo ReplicatedRecoveryBackend::set_recovery_info(
     assert(ssc);
     recovery_info.ss = ssc->snapset;
     auto subsets = crimson::osd::calc_clone_subsets(
-      ssc->snapset, soid, local_missing, last_backfill);
+      ssc->snapset, soid, local_missing, pg.get_info().last_backfill);
```

Comment:
"we may not have info from peer [`fromshard`], use pg's get_info instead." is wrong.  It's not possible for us to be pulling from a peer for which we don't have an info (how would we know the peer has the object in the first place?).

This change does happen to be right because calc_clone_subsets here is being used to figure out which portions of the object need to be pulled from the replica to recover the primary.  That calculation therefore needs last_backfill from the primary, not from the replica.  However, I'm skeptical that this caused a problem because last_backfill on the primary *must* be max for use to have gone active.

Reply:
I amended the commit.